src/validation/permission-exclusions.ts

```diff
@@ -0,0 +1,303 @@
+  //       if (permissionExclusionClash) {
+  //         validationErrors.push(`ERROR: permission '${newPermissions[i]}' can not be assigned`)
+  //         validationFailed = true
+  //       }
+  //     } catch (err) {}
+  //   }
+  //   logger.info(validationErrors)
+  //   if (validationFailed) {
+  //     throw new ValidationError(validationErrors)
+  //   }
+  //   // TODO: We can stop the validation when the first error occurs based on a config param like quickCheck
+  // }
+
+  _getPermissionExclusionsForPermission (
+    permission: string,
+    permissionExclusions: PermissionExclusions[] | PermissionExclusionCombos[]
+  ) : Set<string> {
+    const permissionExclusionsSet: Set<string> = new Set()
+    if (permissionExclusions.length <= 0) {
+      return permissionExclusionsSet
+    }
+    if (isPermissionExclusionCombos(permissionExclusions[0])) {
+      const castedPermissionExclusions = <PermissionExclusionCombos[]>permissionExclusions
+      castedPermissionExclusions
+        .filter(item => item.permissionA === permission)
+        .forEach(perm => permissionExclusionsSet.add(perm.permissionB))
+      castedPermissionExclusions
+        .filter(item => item.permissionB === permission)
+        .forEach(perm => permissionExclusionsSet.add(perm.permissionA))
+    } else {
+      const castedPermissionExclusions = <PermissionExclusions[]>permissionExclusions
+      // eslint-disable-next-line max-len
+      const permissionExclusionsFoundInSetA = castedPermissionExclusions.filter(item => item.permissionsA.includes(permission))
+      // eslint-disable-next-line max-len
+      const permissionExclusionsFoundInSetB = castedPermissionExclusions.filter(item => item.permissionsB.includes(permission))
+      permissionExclusionsFoundInSetA.forEach(item => {
+        item.permissionsB.forEach(perm => permissionExclusionsSet.add(perm))
+      })
+      permissionExclusionsFoundInSetB.forEach(item => {
+        item.permissionsA.forEach(perm => permissionExclusionsSet.add(perm))
+      })
+    }
+    return permissionExclusionsSet
+  }
+
+  validateUserRolePermissions (
+    userRoles: UserRole[],
+    rolePermissions: RolePermissions [],
+    permissionExclusions: PermissionExclusions[] | PermissionExclusionCombos[]
+  ) : void {
+    const validationErrors : string[] = []
+    // Construct a map for role and its permission exclusions
+    const rolePermissionExclusions: Map<string, Set<string>> = new Map()
+    rolePermissions.forEach(role => {
+      // Iterate through permissions and get the set of excluded permissions
+      const permissionExclusionsSet: Set<string> = new Set()
+      // for (let i = 0; i < totalGrantedPermissions.size; i++) {
+      role.permissions.forEach(grantedPerm => {
+        try {
+          // eslint-disable-next-line max-len
+          const permissionExclusionsForPermission = this._getPermissionExclusionsForPermission(grantedPerm, permissionExclusions)
```

Comment:
If we just disable this every time we have a long line, why have it in the first place?
```suggestion
          const permissionExclusionsForPermission = this._getPermissionExclusionsForPermission(
            grantedPerm,
            permissionExclusions
          )
```